### PR TITLE
Xcode 9 fixes

### PR DESCRIPTION
**Pull request by kneth**
## What, How & Why?

We have seen many issue related to Xcode recently. The number of issues increased when Xcode 9 was released but it is still unclear if all the issues are related to that version.

TODO
* [X] The version of `node` inside Xcode must match version used when executing `npm install`
* [x] Find React header files (`React/RCTBridge.h` not found)
* [x] Build React (`libReact.a` not found)
* [x] Fix warnings (`'sendAppEventWithName:body:' is deprecated: Subclass RCTEventEmitter instead`)
* [ ] Update ReactExample to newer UI components (the old navigator is deprecated)
* [ ] Update `CHANGELOG.md`

Status
* [ ] Tests are no longer included in the test app
* [ ] Debug version of the test app crashes with a segmentation fault

### CODE REVIEW

File: src/RealmJS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -887,9 +887,8 @@
 			outputPaths = (
 			);
 			runOnlyForDeploymentPostprocessing = 0;
-			shellPath = /bin/sh;
-			shellScript = "node ../scripts/download-realm.js ios --sync";
-			showEnvVarsInLog = 0;
+			shellPath = "/bin/sh";
+			shellScript = "[ -z \"$NVM_DIR\" ] && export NVM_DIR=\"$HOME/.nvm\"\n\nif [[ -s \"$HOME/.nvm/nvm.sh\" ]]; then\n  . \"$HOME/.nvm/nvm.sh\"\nelif [[ -x \"$(command -v brew)\" && -s \"$(brew --prefix nvm)/nvm.sh\" ]]; then\n  . \"$(brew --prefix nvm)/nvm.sh\"\nfi\nnode ../scripts/download-realm.js ios --sync";
```
Comment (blagoev) [inline src/RealmJS.xcodeproj/project.pbxproj]: this will probably fail on CI

Comment (kneth) [reply src/RealmJS.xcodeproj/project.pbxproj]: why?

Comment (blagoev) [reply src/RealmJS.xcodeproj/project.pbxproj]: it failed last time I committed  it.

File: examples/ReactExample/components/items-screen.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    componentWillMount() {
+        if (Platform.OS == 'ios') {
+            StatusBar.setBarStyle('light-content');
+        }
+    }
+
+    render() {
+        // let objects = realm.objects('Todo');
+        // let extraItems = [
+        //     { name: 'Complete', items: objects.filtered('done = true') },
```
Comment (bdash) [inline examples/ReactExample/components/items-screen.js]: Is there a reason we're adding a bunch of commented-out code (here, and elsewhere)?

Comment (blagoev) [reply examples/ReactExample/components/items-screen.js]: no this is leftover. will remove

File: examples/ReactExample/components/items-screen.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
```
Comment (kneth) [inline examples/ReactExample/components/items-screen.js]: 2017

File: examples/ReactExample/components/styles.js [hunk 4]
```diff
@@ -19,14 +19,15 @@
 'use strict';
 
 import {
-    Navigator,
     Platform,
     StyleSheet
 } from 'react-native';
+// import NavigationExperimental from 'react-native-deprecated-custom-components';
```
Comment (kneth) [inline examples/ReactExample/components/styles.js]: Can be removed (and line 27 too)

File: examples/ReactExample/components/todo-app.js [hunk 5]
```diff
@@ -31,28 +30,44 @@ import {
 
 import TodoItem from './todo-item';
 import TodoListView from './todo-listview';
+import TodoListItem from './todo-list-item';
+import ItemsScreen from './items-screen'
 import realm from './realm';
 import styles from './styles';
 
-export default class TodoApp extends React.Component {
+// import NavigationExperimental from 'react-native-deprecated-custom-components';
```
Comment (kneth) [inline examples/ReactExample/components/todo-app.js]: Can be removed

File: examples/ReactExample/components/todo-app.js [hunk 6]
```diff
@@ -31,28 +30,44 @@ import {
...
         if (this.todoLists.length < 1) {
             realm.write(() => {
-                realm.create('TodoList', {name: 'Todo List', creationDate: new Date()});
+                realm.create('TodoList', { name: 'Todo List', creationDate: new Date() });
             });
         }
         this.todoLists.addListener((name, changes) => {
             console.log("changed: " + JSON.stringify(changes));
+            if (params) {
+                console.error("params.json indicate a test run. Exiting application");
```
Comment (kneth) [inline examples/ReactExample/components/todo-app.js]: "indicate" -> "indicates"

File: examples/ReactExample/components/todo-app.js [hunk 7]
```diff
@@ -31,28 +30,44 @@ import {
...
+                RNExitApp.exitApp();
+            }
         });
         console.log("registered listener");
 
 
         // Bind all the methods that we will be passing as props.
-        this.renderScene = this.renderScene.bind(this);
+        // this.renderScene = this.renderScene.bind(this);
```
Comment (kneth) [inline examples/ReactExample/components/todo-app.js]: Remove?

File: scripts/test.sh [hunk 8]
```diff
@@ -124,126 +133,63 @@ start_packager() {
...
-      sleep 0.05
-    done
-
-    # - Prod `simctl` a few times as sometimes it fails the first couple of times after switching XCode vesions
-    local deadline=$((SECONDS+5))
-    while [ -z "$(xcrun simctl list devices 2>/dev/null)" ] && [ $SECONDS -lt $deadline ]; do
-      : # nothing to see here, will stop cycling on the first successful run
-    done
+  #parse devices
+  IOS_RUNTIME=$(xcrun simctl list runtimes |  grep -m1 -o '(com.apple.CoreSimulator.SimRuntime.iOS.*)' | sed 's/[()]//g')
```
Comment (kneth) [inline scripts/test.sh]: It doesn't work for me locally. Doing `xcrun simctl list runtimes |  grep -m1 'com.apple.CoreSimulator.SimRuntime.iOS.*' | awk '{ print $2 }'`, gets the iOS version number for me locally. When running `./scripts/test.sh react-tests` I see the following in the end:

```
Shuttting down realm-js-test simulator. (device is not deleted. you can use it to debug the app)
Checking tests results
*** ReactTests SUCCESS ***
```

File: tests/react-test-app/index.ios.js [hunk 9]
```diff
@@ -19,23 +19,109 @@
...
 } from 'react-native';
 
+import builder from 'xmlbuilder';
 import React from 'react';
-import { runTests } from './tests';
+import { getTestNames, runTest } from './tests';
+import RNFS from 'react-native-fs';
+import RNExitApp from 'react-native-exit-app-no-history';
+
+// import {setJSExceptionHandler} from 'react-native-exception-handler';
```
Comment (kneth) [inline tests/react-test-app/index.ios.js]: Can this section be removed?

File: scripts/test.sh (hunk 8 above)
Comment (blagoev) [reply scripts/test.sh]: The ending is normal

**Pull request merged**